packages/web3/test/e2e/estimate_gas.test.ts

```diff
@@ -0,0 +1,116 @@
+	};
+	const expectedContractDeploymentTransactionGas =
+		getSystemTestBackend() === 'sepolia' ? 475494 : 475320;
+
+	let web3: Web3;
+
+	beforeAll(() => {
+		web3 = new Web3(provider);
+	});
+
+	afterAll(async () => {
+		await closeOpenConnection(web3);
+	});
+
+	it.each(
+		toAllVariants<{
+			transaction: Transaction;
+			block: 'latest' | 'pending' | 'finalized' | 'safe' | 'blockHash' | 'blockNumber';
+			format: string;
+		}>({
```

Comment:
Most of the tests that read data from the blockchain and are formattable use this `toAllVariants` helper, but being that we already test that data can be formatted as expected in both our unit and integration tests, I don't think this is necessary as it bloats the number of tests significantly (turns a < 100 test to hundreds). It doesn't really hurt to have these tests anyways, it just increases execution time significantly